ts/Stock/Navigator/Navigator.ts

```diff
@@ -540,19 +569,65 @@ class Navigator {
      * @param {Highcharts.NavigatorOptions} options
      *        Options to merge in when updating navigator
      */
-    public update(options: NavigatorOptions): void {
-        // Remove references to old navigator series in base series
-        (this.series || []).forEach((series): void => {
-            if (series.baseSeries) {
-                delete series.baseSeries.navigatorSeries;
+    public update(options: NavigatorOptions, redraw = false): void {
+        const chart = this.chart;
+
+        merge(true, chart.options.navigator, options);
+        this.navigatorOptions = chart.options.navigator || {};
+        this.opposite = pick(
+            this.navigatorOptions.opposite,
+            Boolean(!this.navigatorEnabled && chart.inverted)
+        );
```

Comment:
This looks like a copy&paste. Extract to a method, so we don't have to maintain both places.